pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseSingleStageBrokerRequestHandler.java

```diff
@@ -485,6 +490,16 @@ protected BrokerResponse handleRequest(long requestId, String query, SqlNodeAndO
       if (realtimeTableName == null) {
         realtimeTableConfig = null;
       }
+
+      if ((realtimeTableConfig != null && !_helixAdmin.getResourceIdealState(_clusterName,
```

Comment:
We should avoid looking up ZK in query path. In order to detect disabled tables, we should add a `Set<String>` in `BrokerRoutingManager` to track disabled tables. It can be updated in `processSegmentAssignmentChange()` where we read all ideal states.

Reply:
That makes sense. I'll make the relevant changes